compiler/src/dmd/tokens.d

```diff
@@ -607,7 +607,7 @@ static immutable TOK[TOK.max + 1] Ckeywords =
     with (TOK)
     {
         TOK[TOK.max + 1] tab = identifier;  // default to identifier
-        enum Ckwds = [ auto_, break_, case_, char_, const_, continue_, default_, do_, float64, else_,
+        enum Ckwds = [ auto_, break_, case_, char_, wchar_, dchar_, const_, continue_, default_, do_, float64, else_,
```

Comment:
I suspect the correct way to do this is add `wchar_t`/`char16_t`/`char32_t` as tokens and translate them in the parser, not hack the lexer like this.

Reply:
I considered that, however, these #defines in C. Something like:` #define wchar_t unsigned short` so they won't even enter the D compiler as those types.